test/functional/spec-runner/index.js

```diff
@@ -121,6 +121,21 @@ function generateTopologyTests(testSuites, testContext, filter) {
           afterEach(() => testContext.cleanupAfterSuite());
           testSuite.tests.forEach(spec => {
             it(spec.description, function () {
+              if (requires.authEnabled) {
+                // TODO: We do not have a way to determine if auth is enabled in our mocha metadata
+                // We need to do a admin.command({getCmdLineOpts: 1}) if it errors (code=13) auth is on
+                this.skip();
+              }
+
+              if (
+                spec.operations.some(
+                  op => op.name === 'waitForEvent' && op.arguments.event === 'PoolReadyEvent'
+                )
+              ) {
+                // TODO(NODE-2994): Connection storms work will add new events to connection pool
+                this.skip();
+              }
```

Comment:
We haven't implemented the paused or ready events on a connection pool yet which some tests block on a `'waitForEvent'` operation, until mocha times them out.